src/definitions/ScannedTypehint.hack

```diff
@@ -104,27 +104,29 @@ final class ScannedTypehint {
       $type_name,
     );
 
-    switch ($this->kind) {
-      case ResolvedTypeKind::CALLABLE:
-      case ResolvedTypeKind::GENERIC_PARAMETER:
-        break;
-      case ResolvedTypeKind::QUALIFIED_AUTOIMPORTED_TYPE:
-        if ($options & TypeTextOptions::STRIP_AUTOIMPORTED_NAMESPACE) {
-          $type_name = Str\strip_prefix($type_name, 'HH\\');
-        }
-        break;
-      case ResolvedTypeKind::QUALIFIED_TYPE:
-        if ($relative_to_namespace !== '') {
-          if (Str\starts_with($type_name, $relative_to_namespace.'\\')) {
-            $type_name = Str\strip_prefix(
-              $type_name,
-              $relative_to_namespace.'\\',
-            );
-          } else {
-            $type_name = '\\'.$type_name;
+    if ($this->kind is nonnull) {
```

Comment:
`switch ($this->kind as nonnull)` instead?

Current code assumes that, and we generally consider the need for deeper indentation to be bad for readability

Reply:
I was worried about keeping the current behavior the same. Updated.

Reply:
> Current code assumes that

It is null at runtime... Test failures and exceptions are thrown.
Maybe the author intended to fall through the switch when null was given?